engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/command/SynchronousCommand.scala

```diff
@@ -36,6 +44,9 @@ abstract class SynchronousCommand(maybeRequestId: Option[RequestId])
     }
   }
 
-  def executeSynchronously(implicit ctx: RuntimeContext): Unit
+  def executeSynchronously(implicit
+    ctx: RuntimeContext,
+    ec: ExecutionContext
+  ): Unit
```

Comment:
Sorry for post-mortem comment, but I haven't seen this before.

---

Why is the argument `ec: ExecutionContext` added here??

If I see correctly - it is not used anywhere anyway.

And it seems rather dangerous - the computation performed by this overload is supposed to happen synchronously, so we should not need an `ExecutionContext` for it, right?

Reply:
Because of `PushContextCmd`. It needs to be in the same execution queue as open/close commands because there is a chance that client will issue it *before* it has a chance to run, resulting in some obscure NPEs. This is not a hypothetical solution and was quite frequent in our tests on slower machines.

Having said that, I'm going to rewrite it a bit so that open/close commands actually issue replies. That way we can ensure the correct order of execution, and then drop `PushContextCmd` from synchronous commands.